python/deltalake/writer.py

```diff
@@ -302,7 +302,7 @@ def __enforce_append_only(
 class DeltaJSONEncoder(json.JSONEncoder):
     def default(self, obj: Any) -> Any:
         if isinstance(obj, bytes):
-            return obj.decode("unicode_escape")
+            return obj.decode("unicode_escape", "ignore")
```

Comment:
Based on [the protocol](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#partition-value-serialization), I think we actually want to escape with `\u`:
```suggestion
            return obj.decode("unicode_escape", "backslashreplace")
```